File: admin/src/app/models/update-requests/discount-update-request.ts
import { AutoMap } from "@automapper/classes"
import { BaseUpdateRequest } from ".."

export class DiscountUpdateRequest extends BaseUpdateRequest {
    @AutoMap()
    id?: number;
    @AutoMap()
    percentDiscount: number;
    @AutoMap()
    maxDiscountPrice: number;
    @AutoMap()
    fromDate: Date;
    @AutoMap()
    toDate: Date;
}

File: App.Models/DTOs/UpdateRquests/DiscountUpdateRequest.cs
using System;
using App.Common.Model.DTOs;

namespace App.Models.DTOs.UpdateRquests
{
    public class DiscountUpdateRequest : BaseUpdateRequest<int>
    {
        public int? PercentDiscount { get; set; }
        public double? MaxDiscountPrice { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
    }
}
